src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {
+    View,
+    Text,
+    TouchableOpacity,
+    ActivityIndicator,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+
+const propTypes = {
+    // Callback to inform parent modal of success
+    onStepComplete: PropTypes.func.isRequired,
+
+    // is page content currently being retrieved
+    isLoading: PropTypes.bool.isRequired,
```

Comment:
Would it probably be better to just connect this component directly to Onyx?

Reply:
Makes sense, but I wonder whether this should apply in the case of a controlled component. 

I don't think it makes a huge difference either way, but might we prefer to have ALL state managed by the component, rather than some from the controlling component, and some directly from Onyx?

Reply:
My thought is that if a React component is written as a reusable component then it would follow these guidelines:

1. It lives in the `component` folder
2. It has as few dependencies as possible
3. It probably would NOT connect to Onyx (only 3 of the ~50 components we have now connect to Onyx)

If a React component is written as a non-reusable component (views, pages, etc.) then it would follow these guidelines:

1. It lives outside the `components` folder as appropriate
2. It can have as many dependencies as it needs
3. Any data that is stored in Onyx should NOT be passed as props and should always be accessed using `withOnyx`

I don't know that this has really be talked about a whole lot, but that's what is in my mind. I'd like to invite @marcaaron to the conversation to see what his ideas are about this as well.

Reply:
Thanks for sharing, that sounds like good guidelines to document and follow.

Reply:
Cool, I moved it over to SO for any further discussion ([link](https://stackoverflow.com/c/expensify/questions/8486/8487#8487)).

Reply:
Implemented, [here's the commit ](https://github.com/Expensify/Expensify.cash/pull/1561/commits/ccb18e9861dfcae0d7f3b2ffd92e56b0b48bc606)for reference